src/components/AnimateSharedLayout/stack.ts

```diff
@@ -241,3 +248,62 @@ export class LayoutStack {
         return this.lead?.config.transition
     }
 }
+
+export class TreeStack extends Stack<LayoutTree> {
+    sortOrder() {
+        this.order.sort((a, b) => a.props.layoutOrder - b.props.layoutOrder)
+    }
+
+    add(child: LayoutTree) {
+        if (child.layoutOrder === undefined) return
+
+        const { layoutOrder } = child.props
+        let index = (this.order as LayoutTree[]).findIndex(
+            stackChild => layoutOrder <= (stackChild.layoutOrder || 0)
+        )
+
+        if (index === -1) index = this.order.length
+
+        this.order.splice(index, 0, child)
+    }
+
+    /**
+     * Generate LayoutStacks from the children of the Lead and Follow trees.
+     */
+    createLayoutStacks(): Map<string, LayoutStack> {
+        const stacks: Map<string, LayoutStack> = new Map()
+
+        if (this.follow) {
+            for (const child of this.follow.children) {
+                if (!child.layoutId) continue
+
+                const stack = new LayoutStack()
+
+                child.isPresent = this.follow.isPresent
+                stack.add(child)
+                stacks.set(child.layoutId, stack)
+                stack.updateLeadAndFollow()
```

Comment:
Maybe if we could instantiate LayoutStack with a child that would set the single child to the lead and avoid having to call `updateLeadAndFollow` early here? Essentially I just want to ensure that the stacks created for layoutId's that aren't present in the lead tree are setup right

Alternatively maybe this isn't even necessary, if there is no follow does the stack even animate and need to have a lead? 🤔